demos/camera_demo/src/main.cpp

```diff
@@ -267,8 +304,12 @@ int main(int argc, char *argv[])
 
     delete signalThread;
     delete videoSensor;
-    delete frameProcessor;
-    delete orientationInput;
+    for (FrameProcessor* frameProcessor : frameProcessors) {
+        delete frameProcessor;
+    }
+    if (orientationInput != colorTracking) {
```

Comment:
This feels awkward. What is `orientationInput` when its not doing color tracking?

Reply:
The orientationInput can be either color tracking, or the keyboard, or the accelerometer. Only the color tracking is a Frame processor as well as an OrientationInput. There is a separate issue which is that the current design can only handle a single OrientationInput. So you can't use both the keyboard and color tracking, for example.

I agree line 310 is awkward. But if we don't do it then there is a double free on the colorTracking. We could probably wrap these in some C++ smart pointers to solve this problem.